code_map.py

```diff
@@ -828,12 +860,34 @@ def on_text_command(self, view, command_name, args):
                     f = False if CodeMapListener.navigating else True
                     navigate_to_line(view, give_back_focus=f)
                     return ("code_map_select_line", None)
+        return None
+
+    # -----------------
+
+    def on_window_command(self, window, command_name, args):
+        '''Very unstable switching projects, safety measure'''
+        global TEMP_VIDS, TEMP_VIEWS, CURRENT_TEMP_ID, ACTIVE
+
+        reset = ["prompt_open_project_or_workspace",
+                 "prompt_switch_project_or_workspace",
+                 "prompt_select_workspace",
+                 "open_recent_project_or_workspace",
+                 "close_workspace",
+                 "project_manager"]
+
+        if ACTIVE and command_name in reset:
+
+            reset_globals()
+
+            # if command_name == "project_manager":
+            #     window.run_command('show_code_map')
+        return None
 
```

Comment:
If project is switched, CodeMap is inactivated(but not closed). When a new project is loaded, if the panel is open it must be synched first.